test/nodes/test_circuit_breaker.py

```
@@ -0,0 +1,646 @@
+import asyncio
+import os
+import sys
+import threading
+import time
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+# ---------------------------------------------------------------------------
+# Path setup – ensure nodes/src/nodes is importable
```

Ruff (0.15.7)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)